app/controllers/pets_controller.rb

```diff
@@ -30,25 +36,37 @@ def update
       description: params[:pet_description],
       approximate_age: params[:pet_approximate_age],
       sex: params[:pet_sex])
-    redirect_to "/pets/#{params[:id]}"
+      if pet.save
+        flash[:success] = "The pet has been updated."
+        redirect_to "/pets/#{params[:id]}"
+      else
+        flash[:error] = pet.errors.full_messages.to_sentence
+        redirect_to "/pets/#{pet.id}/edit"
+      end
   end
 
   def destroy
+    favorites.remove_pet(params[:id])
+    session[:favorites] = favorites.pets_favorite
+
     Pet.destroy(params[:id])
     redirect_to '/pets'
   end
 
+
   def change_adoption
     pet = Pet.find(params[:petid])
     if pet.adoption_status == "adoptable"
       pet.update(adoption_status: "pending")
       pet_app = PetApplication.where(pet_id: params[:petid], application_id: params[:appid])[0]
       pet_app.update(adopted: true)
+      pet.update(approved: true)
```

Comment:
This confirms that the pet has been approved. There is an approved method in PetApplication but I wanted to be able to call this from pet.